rust/noosphere-sphere/src/context.rs

```diff
@@ -254,20 +111,30 @@ where
         unimplemented!()
     }
 
-    /// Resolve the most recent version in the local history of the sphere
-    pub async fn head(&self) -> Result<Cid> {
-        let sphere_identity = self.identity();
-        self.db
-            .get_version(&self.sphere_identity)
-            .await?
-            .ok_or_else(|| anyhow!("No version found for {}", sphere_identity))
-    }
-
     /// The identity of the sphere
     pub fn identity(&self) -> &Did {
         &self.sphere_identity
     }
 
+    /// The identity of the counterpart sphere in use during this session, if
+    /// any; note that this will cause a request to be made to a gateway if no
+    /// handshake has yet occurred.
+    pub async fn counterpart_identity(&self) -> Result<Did> {
```

Comment:
How is `counterpart_identity()` used? Is this the gateway's counterpart, e.g. *this* sphere's identity?

Reply:
Strictly speaking the counterpart is relative to your sphere. For the gateway, the counterpart is the user's sphere. For the user, the counterpart is the gateway's sphere.

That said, this method isn't in use and its implementation only works for the user, so we should just remove it.